Cargo.toml

```diff
@@ -12,11 +12,13 @@ repository = "https://github.com/johanhelsing/bevy_pkv"
 [dependencies]
 thiserror = "1.0"
 serde = "1.0"
+rocksdb = "0.20.1"
 bevy_ecs = "0.10" # we need for deriving Resource in PkvStore
 
 [features]
 default = ["bevy"]
```

Comment:
Hmm... Is there a way we could make default features enable sled on native, but not also break wasm builds?

Reply:
This concern was fixed with `rocksdb = { version = "0.20", optional = true }` above correct?

Reply:
well, it does, but it also means sled will still be a dependency when the `rocksdb` feature is enabled... `sled` should somehow be an optional dependency.

Reply:
We could make `local_storage` a feature as well, but that would require people to manually select the backend they would like, which would be more annoying to use.

Reply:
https://stackoverflow.com/questions/67373380/can-i-set-cargo-projects-default-features-depending-on-the-platform

Is this a solution?

Reply:
The OP in that Stack Overflow post didn't seem to think it was a good solution so maybe not...

Reply:
yeah, that's actually what we're currently doing, but now we're running into the issues mentioned in the last comment: https://stackoverflow.com/questions/67373380/can-i-set-cargo-projects-default-features-depending-on-the-platform#comment124334292_70316751

Reply:
I think most crates just make native work, and then tell their wasm users to deal with it by removing default-features and adding a default feature.

At least that what `surf` does: https://github.com/http-rs/surf/blob/main/Cargo.toml

Could maybe check some more cross-platform crates for inspiration.

Reply:
That could work. Or worse comes to worse maybe I make the RocksDB version a fork of your project and we just link each other on the README files? Then I would just re-implement any changes and updates you push from sled to rocksdb when needed?

Once again that would be worse case scenario. It's up to you how you want to do it. I don't want to come in and change how you do things after you have put the work in time into this library.

Reply:
winit solves this by having unsupported features just be no-ops on platforms where they're not supported. e.g. if you add the x11 or wayland features on wasm, they do nothing. They do this by addign a `build.rs`: https://github.com/rust-windowing/winit/blob/fbea75d31f4025825e9bf88b99d7c92e4eb79273/build.rs#L47

We could probably do something similar.

Reply:
Sure, sounds good to me

Reply:
I forgot to ask, do you want me to create the build.rs file or are you doing it?

Reply:
I don't have time to do it today at least. Feel free to give it a shot 👍

Reply:
So, I've never worked with WASM before and I can't even get the tests to run on the unaltered main branch of your code.

`cargo test --target wasm32-unknown-unknown` doesn't work. I'm not sure if I should be writing a build.rs to make WASM work if I can't even figure out how to run a WASM test.

What am I doing wrong?

Reply:
Running tests for wasm is often a bit tricky. For this project i think i just settled for checking that the examples work. What i do is install wasm-server-runner and configure it to be the runner for wasm globally.

Then you can just do

    cargo run --example basic --target wasm32-unknown-unknown

And it should work

Reply:
![image](https://user-images.githubusercontent.com/43654357/226213272-981faab0-2027-4238-b515-4f5507ff81e5.png)

I did what you said with the current main branch and I'm getting this error when it runs

Reply:
Thanks, it seems I was a bit quick when I did the bevy 0.10 port. Will be fixed in #22

Reply:
I created a build.rs script like you asked. I think everything works, but please feel free to clone my fork and double check.

Reply:
Looks good to me :) Thanks!